heim-net/src/sys/windows/nic.rs

```diff
@@ -1,57 +1,284 @@
 use heim_common::prelude::*;
 
+use std::net::Ipv4Addr;
+use std::net::Ipv6Addr;
+use std::net::SocketAddrV4;
+use std::net::SocketAddrV6;
+
+use std::ffi::CStr;
+use widestring::UCStr;
+
+use winapi::shared::minwindef::ULONG;
+use winapi::shared::ntdef::NULL;
+use winapi::shared::ws2def::SOCKET_ADDRESS;
+use winapi::shared::ws2ipdef::SOCKADDR_IN6;
+use winapi::shared::ws2def::{AF_INET, AF_INET6};
+use winapi::shared::ws2def::AF_UNSPEC;
+use winapi::shared::winerror::{ERROR_BUFFER_OVERFLOW, NO_ERROR};
+use winapi::um::iphlpapi::GetAdaptersAddresses;
+use winapi::um::iptypes::GAA_FLAG_INCLUDE_PREFIX;
+use winapi::um::iptypes::PIP_ADAPTER_ADDRESSES;
+use winapi::um::iptypes::IP_ADAPTER_ADDRESSES;
+
 use crate::Address;
 
-#[derive(Debug)]
-pub struct Nic;
+
+
+#[derive(Clone, Debug)]
+pub struct Nic {
+    index: u32,
+    guid: String,
+    friendly_name: String,
+    address: Option<Address>,
+    netmask: Option<Address>,
+}
+
+
+fn sockaddr_to_ipv4(sa: SOCKET_ADDRESS) -> Option<Address> {
+    // Check this sockaddr can fit one short and a char[14]
+    // (see https://docs.microsoft.com/en-us/windows/win32/winsock/sockaddr-2)
+    // This should always happen though, this is guaranteed by winapi's interface
+    if sa.iSockaddrLength < 2+14 {
```

Comment:
Can we replace `2 + 14` with `std::mem::size_of::<SOCKADDR>()` as in `sockaddr_to_ipv6` below?
It will be more clear what the intent in here.

Reply:
You're right! I've just corrected it